packages/test/test-utils/src/container-runtime-factories/containerRuntimeFactoryWithDefaultDataStore.ts

```diff
@@ -0,0 +1,97 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { IContainerRuntimeOptions } from "@fluidframework/container-runtime/internal";
+import type {
+	IContainerRuntime,
+	// eslint-disable-next-line import/no-deprecated
+	IContainerRuntimeWithResolveHandle_Deprecated,
+} from "@fluidframework/container-runtime-definitions/internal";
+import type { FluidObject, IRequest, IResponse } from "@fluidframework/core-interfaces";
+import type {
+	IFluidDataStoreFactory,
+	NamedFluidDataStoreRegistryEntries,
+} from "@fluidframework/runtime-definitions/internal";
+import { RequestParser } from "@fluidframework/runtime-utils/internal";
+
+import { BaseContainerRuntimeFactory } from "./baseContainerRuntimeFactory.js";
+
+const defaultDataStoreId = "default";
+
+async function getDefaultFluidObject(runtime: IContainerRuntime): Promise<FluidObject> {
+	const entryPoint = await runtime.getAliasedDataStoreEntryPoint("default");
+	if (entryPoint === undefined) {
+		throw new Error("default dataStore must exist");
+	}
+	return entryPoint.get();
+}
+
+/**
+ * {@link ContainerRuntimeFactoryWithDefaultDataStore} construction properties.
+ */
+export interface ContainerRuntimeFactoryWithDefaultDataStoreProps {
+	defaultFactory: IFluidDataStoreFactory;
+	/**
+	 * The data store registry for containers produced.
+	 */
+	readonly registryEntries: NamedFluidDataStoreRegistryEntries;
+
+	/**
+	 * The runtime options passed to the IContainerRuntime when instantiating it
+	 */
+	readonly runtimeOptions?: IContainerRuntimeOptions;
+
+	/**
+	 * Function that will initialize the entryPoint of the IContainerRuntime instances
+	 * created with this factory
+	 */
+	readonly provideEntryPoint?: (runtime: IContainerRuntime) => Promise<FluidObject>;
+}
+
+/**
+ * A ContainerRuntimeFactory that initializes Containers with a single default data store, which can be requested from
+ * the container with an empty URL.
+ */
+export class ContainerRuntimeFactoryWithDefaultDataStore extends BaseContainerRuntimeFactory {
```

Comment:
Would we want to preemptively mark any of these as deprecated?  E.g. I could imagine that there's a path to removal if:

1. createSumarizerFromFactory makes its param non-optional
2. We do ...something with LocalCodeLoader?  fluidEntryPoint is old tech anyway so it's not a great pattern overall.

Reply:
I would be in favor of making these at least _externally_ deprecated. Migrating usages of this stuff is beyond the scope of what I have time to do right now, but I am more than happy to slap `@deprecated` tags on the types if we want to 😜

Reply:
I can do that. Migrating usages is beyond the scope of what I have time for right now, but I will happily mark them as deprecated if we have reasonable alternatives we can point to.